workers/price-reporter/src/manager.rs

```diff
@@ -1,20 +1,63 @@
 //! Defines the PriceReporterExecutor, the handler that is responsible
 //! for executing individual PriceReporterJobs.
-use common::default_wrapper::{DefaultOption, DefaultWrapper};
-use common::types::exchange::PriceReporterState;
-use common::types::token::Token;
-use common::types::CancelChannel;
-use common::{new_async_shared, AsyncShared};
-use job_types::price_reporter::{PriceReporterJob, PriceReporterReceiver};
-use std::{collections::HashMap, thread::JoinHandle};
-use tokio::runtime::Runtime;
-use tokio::sync::oneshot::Sender as TokioSender;
-use tracing::{error, info, info_span, warn, Instrument};
-use util::err_str;
-
-use crate::errors::{ExchangeConnectionError, PriceReporterError};
-
-use super::{reporter::Reporter, worker::PriceReporterConfig};
+
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    thread::JoinHandle,
+};
+
+use atomic_float::AtomicF64;
+use common::types::{
+    exchange::{Exchange, PriceReport, PriceReporterState},
+    token::Token,
+    Price,
+};
+use statrs::statistics::{Data, Median};
+use util::get_current_time_seconds;
+
+use crate::{errors::PriceReporterError, worker::PriceReporterConfig};
+
+// pub mod external_executor;
+pub mod native_executor;
+
+// -------------
+// | Constants |
+// -------------
+
+/// If Binance has not reported an update within
```

Comment:
nit: I'd avoiding wording these types of things as "if binance...", and keep it abstract. While currently we use Binance prices the code is written more extensibly than that and our language should reflect that!